### PR TITLE
Photo Resistor Helper Unit Tests Added

### LINKED ISSUE #308: Write Unit Tests for PhotoResistorHelper
**Description**
The PhotoResistorHelper class manages the functionality of a photoresistor sensor using Pi4J's DigitalInput and DigitalOutput components. Unit tests are required to validate its behavior, including sensor activation, darkness level detection, event listener management, and logging.

**Develper Notes:** 
1. Test Initialization Logic:
  
  Verify that the photoResistorOutput is set to high during initialization.
  Ensure that sensorActive is set to true.
  Check that isDark is initialized correctly based on the photoResistorInput state.

2. Test Darkness Level Calculation (updateDark Method):
  
  Simulate sensor state changes and verify that darknessValue is updated correctly.
  Validate that isDark is set based on the darknessThreshold.

3. Test getDark Method:

Confirm that it returns the correct darknessValue.

4. Test setToLow Method:

Ensure that photoResistorOutput is set to low when photoResistorInput is high.

5. Test setDarknessThreshold Method:

Verify that darknessThreshold is updated correctly and the expected log message is generated.

6. Test Event Listener Management:

  Validate that addEventListener correctly adds an event listener to photoResistorInput.
  Check that the listener is triggered when the input state changes.
  Ensure that removeEventListener removes the listener and sets sensorActive to false.

7. Mocking Dependencies:

  Use Mockito to mock DigitalInput, DigitalOutput, and DigitalStateChangeListener.
  Ensure unit tests focus on the class behavior without requiring real hardware.

8. Edge Cases:. 
  Test behavior when sensorActive is false.
  Verify the initialize method does not start the scheduled task if sensorActive is false.
  Ensure that multiple calls to removeEventListener do not cause errors.

Acceptance Criteria

- [ ] All test cases outlined in the tasks above are implemented.

- [ ] Code coverage for the MicroSwitchHelper class is at least 70%.

- [ ] Tests are written using a Java JUnit testing framework

- [ ] Mocking is implemented using a framework such as Mockito or a similar tool.

- [ ] All tests pass successfully, and edge cases are handled gracefully.

### PR DESCRIPTION
**Pull request by avaenk**
Fixes #308 

What was changed: implemented unit tests for the photo resistor helper class to validate its behavior, including sensor activation, darkness level detection, event listener management, and logging. Included test coverage to verify the correct state transitions and expected behaviors. Used mocking to simulate hardware so the tests do not entirely rely on devices. 

Why it was changed: To ensure that the photo resistor helper class functions correctly under many scenarios, including normal operations and edge cases. It also improves test coverage and validate sensor state transitions, ensuring the cass meets project requirements for testing. 

How it was changed: Verified that photo resistor output is set to high during initialization and that isDark is set correctly based on the input state. Simulated sensor state changes and validated the darkness value updates correctly based on the darkness threshold. Ensured event listeners are properly added and removed using mocks for digital input. Using mokcito to mock digital input and digital output and digital state change listener allowing unit testing to be independent of hardware. Verified correct behavior when sensor active is false, ensured initialize does not start the scheduled task if sensor active is false, and validated that multiple calls to remove event listener do not cause errors.

### CODE REVIEW

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
The use of Thread.sleep(600); in PhotoresistorHelperTest introduces unnecessary delays, making the tests slower and potentially unreliable. Instead of Thread.sleep, consider using dependency injection to mock System.currentTimeMillis() or use Mockito.when(System.currentTimeMillis()).thenReturn(...) to simulate time changes. This will make the tests more efficient and remove unnecessary wait times. Please update this accordingly. The rest of the test structure looks solid! 👍

File: pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PhotoResistorHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import com.opensourcewithslu.mock.MockDigitalInput;
+import com.pi4j.io.gpio.digital.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.lang.reflect.Field;
+
+
+public class PhotoResistorHelperTest{
```
Comment (RalphTan37) [inline pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/PhotoResistorHelperTest.java:15]: Each function looks properly structured but would like to see some documentation notes to explain how some components work. Other than that, these unit tests look great!

**Review by yrlmanoharreddy — APPROVED** [review]
Great work on the unit tests for PhotoResistorHelper! ✅ The tests are well-structured, cover the necessary scenarios, and ensure reliability. Everything looks good to me—approved!
